pilot/pkg/model/push_context.go

```diff
@@ -757,6 +779,57 @@ func (ps *PushContext) initServiceAccounts(env *Environment, services []*Service
 	}
 }
 
+// Caches list of authentication policies
+func (ps *PushContext) initAuthnPolicies(env *Environment) error {
+	authNPolicies, err := env.List(schemas.AuthenticationPolicy.Type, NamespaceAll)
+	if err != nil {
+		return err
+	}
+
+	sortConfigByCreationTime(authNPolicies)
+	ps.AuthnPolicies = processedAuthnPolicies{
+		policies: map[host.Name][]*authnPolicyByPort{},
+	}
+
+	for _, spec := range authNPolicies {
+		policy := spec.Spec.(*authn.Policy)
+		if len(policy.Targets) > 0 {
+			for _, dest := range policy.Targets {
+				hostName := ResolveShortnameToFQDN(dest.Name, spec.ConfigMeta)
+				if len(dest.Ports) > 0 {
+					for _, port := range dest.Ports {
+						ps.addAuthnPolicy(hostName, port, policy)
+					}
+				} else {
+					ps.addAuthnPolicy(hostName, nil, policy)
+				}
+
+			}
+		} else {
+			// if no targets provided, store at namespace level
+			ps.addAuthnPolicy(host.Name(spec.Namespace), nil, policy)
```

Comment:
Is this correct if we're sharing the same map for both namespace policy and host name as the key? I saw we have `ResolveShortNameToFQDN` invocation above. Are we basically betting that, customers won't create a namespace as "foo.svc.cluster.local"?

Reply:
if a customer having a namespace `foo.svc.cluster.local` is a valid concern, I can definitely separate them into two separate maps, one for FQDN and another for namespace.  @diemtvu  @rshriram @linsun thoughts?

Reply:
you can't create a namespace with a `.`

Reply:
I agree its confusing/wrong to use namespace as a `host.Name` but they can't actually get a name conflict as far as I can tell?